src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -114,13 +88,14 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
     }
 
     const mode = stateObj.state in modeIcons ? stateObj.state : "unknown-mode";
+
     return html`
-      ${this.renderStyle()}
       <ha-card
         class="${classMap({
           [mode]: true,
-          large: this._broadCard!,
-          small: !this._broadCard,
+          large: this.clientWidth > 450,
+          medium: this.clientWidth <= 450 && this.clientWidth >= 350,
+          small: this.clientWidth < 350,
```

Comment:
Is there much benefit in only having this in the initial loading of the card?

Reply:
IIRC all cards are rebuilt on large layout changes, i.e. when the screen size changes enough to adjust the columns. But you may want to rescale even on smaller changes, perhaps?

Reply:
This is now checked at every render, but not necessarily on a resize of the window because we only render on a change of properties. clientWidth is not a property.